src/include/elektra.h

```diff
@@ -0,0 +1,466 @@
+	static const ELEKTRA_TAG (typeName) ELEKTRA_TAG_NAME (tagName) = {                                                                 \
+		(keyname),                                                                                                                 \
+		ELEKTRA_GET_BY_TAG (typeName),                                                                                             \
+		ELEKTRA_GET_ARRAY_ELEMENT_BY_TAG (typeName),                                                                               \
+		ELEKTRA_SET_BY_TAG (typeName),                                                                                             \
+		ELEKTRA_SET_ARRAY_ELEMENT_BY_TAG (typeName),                                                                               \
+	};
+
+// endregion
+
+// region Tags for built-in types
+/**************************************
+ *
+ * Tags for built-in types
+ *
+ **************************************/
+
+ELEKTRA_TAG_DECLARATIONS (const char *, String)
+ELEKTRA_TAG_DECLARATIONS (kdb_boolean_t, Boolean)
+ELEKTRA_TAG_DECLARATIONS (kdb_char_t, Char)
+ELEKTRA_TAG_DECLARATIONS (kdb_octet_t, Octet)
+ELEKTRA_TAG_DECLARATIONS (kdb_short_t, Short)
+ELEKTRA_TAG_DECLARATIONS (kdb_unsigned_short_t, UnsignedShort)
+ELEKTRA_TAG_DECLARATIONS (kdb_long_t, Long)
+ELEKTRA_TAG_DECLARATIONS (kdb_unsigned_long_t, UnsignedLong)
+ELEKTRA_TAG_DECLARATIONS (kdb_long_long_t, LongLong)
+ELEKTRA_TAG_DECLARATIONS (kdb_unsigned_long_long_t, UnsignedLongLong)
+ELEKTRA_TAG_DECLARATIONS (kdb_float_t, Float)
+ELEKTRA_TAG_DECLARATIONS (kdb_double_t, Double)
+
+#ifdef ELEKTRA_HAVE_KDB_LONG_DOUBLE
+
+ELEKTRA_TAG_DECLARATIONS (kdb_long_double_t, LongDouble)
+
+#endif // ELEKTRA_HAVE_KDB_LONG_DOUBLE
+
+ELEKTRA_TAG_DECLARATIONS (int, Enum)
+
+// endregion Tags for built-in types
+
+// region Basics
+/**************************************
+ *
+ * Basics
+ *
+ **************************************/
+
+Elektra * elektraOpen (const char * application, KeySet * defaults, ElektraError ** error);
+void elektraClose (Elektra * elektra);
+
+// endregion Basics
+
+// region Error-Handling
+/**************************************
+ *
+ * Error-Handling
+ *
+ **************************************/
+
+void elektraFatalErrorHandler (Elektra * elektra, ElektraErrorHandler fatalErrorHandler);
+
+// endregion
+
+// region Helpers for code generation
+/**************************************
+ *
+ * Helpers for code generation
+ *
+ **************************************/
+
+Key * elektraFindKey (Elektra * elektra, const char * name, KDBType type);
+Key * elektraFindArrayElementKey (Elektra * elektra, const char * name, size_t index, KDBType type);
+void elektraFatalError (Elektra * elektra, ElektraError * fatalError);
+
+// endregion Helpers for code generation
+
+// region Getters
+/**************************************
+ *
+ * Getters
+ *
+ **************************************/
+
+const char * elektraGetValue (Elektra * elektra, const char * name);
```

Comment:
should be const  void *

Reply:
The function calls `keyString()` internally, so it has the same return type.

If the function should support binary data, I would propose this signature instead:
```c
const void * elektraGetValue (Elektra * elektra, const char * name, size_t * size);
```
Passing `NULL` to size would be allowed, that way one could use `(const char *) elektraGetValue (elektra, "mykey", NULL);` to get the current behaviour.

Reply:
@markus2330 any comment?

Reply:
Sorry for the delay.

I would recommend to not support binary data for now. So simply remove the `*Value*` functions

Reply:
But your proposal is also ok. (If you have already implemented, documented and tested it.)